resources/js/chat/chat-state-store.ts

```diff
@@ -213,12 +256,17 @@ export default class ChatStateStore implements DispatchListener {
 
   @action
   private handleChatChannelJoinEvent(event: ChannelJoinEvent) {
-    this.channelStore.update(event.json);
-
-    if (this.waitJoinChannelUuid != null && this.waitJoinChannelUuid === event.json.uuid) {
-      this.selectChannel(event.json.channel_id);
-      this.waitJoinChannelUuid = null;
-      if (event.json.type === 'ANNOUNCE') {
+    const json = event.json;
+    this.channelStore.update(json);
+
+    if (this.waitAddChannelType === 'announce' && this.waitAddChannelId === json.uuid
+      || this.waitAddChannelType === 'existing' && this.waitAddChannelId === json.channel_id
```

Comment:
isn't it enough to just check type of waitAddChannelId and match the uuid/channel_id accordingly? That way the waitAddChannelType wouldn't be needed

(which also a bit confusing why waitAddChannelId being a string means it's announce type)